symengine/visitor.cpp

```diff
@@ -201,6 +201,31 @@ void TransformVisitor::bvisit(const MultiArgFunction &x)
     result_ = nbarg;
 }
 
+void TransformVisitor::bvisit(const Piecewise &x)
+{
+    auto branch_cond_pairs = x.get_vec();
+    PiecewiseVec new_pairs;
+    bool changed = false;
+    for (const auto &branch_cond : branch_cond_pairs) {
+        auto branch = branch_cond.first;
+        auto cond = branch_cond.second;
+        auto new_branch = apply(branch);
+        // decltype(cond) new_cond =
+        // rcp_static_cast<some-kind-of-boolean-cse-symbol?>(apply(rcp_static_cast<const
+        // Basic>(cond)));
```

Comment:
I couldn't figure out how to do subexpression elimination on the conditionals, so before merging I should probably just delete these comments...